### PR TITLE
Added in the parallax starfield

### LINKED ISSUE #53: Game has a randomly created starfield

### LINKED ISSUE #54: Stars scroll from right to left, randomly respawning at some Y on the right edge after exiting the screen

### LINKED ISSUE #55: Stars move at different rates, giving a parallax effect

### LINKED ISSUE #56: Stars are in different colors
Star colors are evenly distributed

### PR DESCRIPTION
**Pull request by dhowerton**
Closes #53, closes #54, closes #55, closes #56.

I chose five non-white, non-red colors for the starfield (yellow, aqua, chartreuse, violet, and pink).  Used the alpha property to give those colors some range.  Stars "far" in the background, moving slowly are dimmer than the rest.

I changed the enemy projectile image to the purest red there is for contrast.

I also added the occasional "star streak" for some visual flavor.

Playable build here: http://mocsarcade.github.io/starjunk/82/

### CODE REVIEW

**Comment by ehgoodenough** [comment]
What's the thought on the colors used here? I had thought the stars would be the same colors we're using for the rest of our color scheme.

**Comment by dhowerton** [comment]
That would leave us with just yellow, blue, and green.  Trying to play with white and/or red confused the issue of what was a projectile.  The pink and purple was just to add more variety to the scheme, but they are easily removed.

**Comment by dhowerton** [comment]
Created a 1x1 image for the stars.  Starfield moves at a slightly lower speed.  Star streaks happen a little less frequently.  Removed purple and pink.

I left the max star count the same as it didn't look too busy with the smaller star size.  Let me know what you think.

**Comment by ehgoodenough** [comment]
Thanks!! Did you update the playable link?

**Comment by dhowerton** [comment]
Yeah, I updated it.

File: source/StarStreak.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+var Pixi = require("pixi.js")
+
+import Reference from "./Reference.js"
+import Utility from "./Utility.js"
+
+export default class StarStreak extends Pixi.Sprite {
+    constructor(parallax) {
```
Comment (Phyoog) [inline source/StarStreak.js]: It doesn't look like `parallax` is getting used in `StarStreak`. No sense passing it to the constructor if it's unneeded.

File: source/Star.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        this.tint = Reference.STAR_COLORS[Math.floor(Math.random() * Reference.STAR_COLORS.length)]
+        this.alpha = parallax * .1
+        this.speed = parallax * .4
+    }
+
+    update(delta) {
+        this.position.x -= this.speed
+        if (this.position.x < -this.width) {
+            this.position.x *= -1
+            this.position.x += Reference.GAME_WIDTH
```
Comment (codaventure) [inline source/Star.js]: This is an interesting way to cycle the stars back to the right side of the screen. What's the *= -1 for? Looks like it moves the x position to 1, then adds the game width to it. Why not just set the position to the game width?

**Comment by codaventure** [comment]
Much better with the 1x1 stars! A lot easier to discern between stars and projectiles now.

**Comment by Phyoog** [comment]
Looks good. Merging.

**Pull request merged**